testing/execution.ts

```diff
@@ -951,3 +941,151 @@ class DeveloperError extends Error {
+ * @param params The params to pass to the formula
+ * @param manifest The manifest of the pack
+ * @param executionContext The execution context
+ * @param bundleSourceMapPath The source map path
+ * @param bundlePath The bundle path
+ * @param maxRows The max rows to sync
+ * @param vm Whether to run in a VM
+ * @param contextOptions The context options.
+ * @param contextOptions.syncOptions.includePermissions Whether to include permissions in the sync table rows.
+ *    Will return an empty array if this is false.
+ * @returns Returns an object with result (the sync table rows) and rowAccessDefinitions (if we opt into it)
+ */
+async function executeSyncFormulaWithContinuations({
+  formulaSpecification,
+  params,
+  manifest,
+  executionContext,
+  bundleSourceMapPath,
+  bundlePath,
+  maxRows = DEFAULT_MAX_ROWS,
+  vm,
+  contextOptions,
+}: {
+  formulaSpecification: SyncFormulaSpecification;
+  params: string[];
+  manifest: BasicPackDefinition;
+  executionContext: SyncExecutionContext;
+  bundleSourceMapPath: string;
+  bundlePath: string;
+  maxRows?: number;
+  vm?: boolean;
+  contextOptions?: ContextOptions;
+}) {
+  let result = [];
+  let rowAccessDefinitions = [];
+  let iterations = 1;
+  do {
+    if (iterations > MaxSyncIterations) {
+      throw new Error(
+        `Sync is still running after ${MaxSyncIterations} iterations, this is likely due to an infinite loop.`,
+      );
+    }
+    const response: GenericSyncFormulaResult = vm
+      ? await executeFormulaOrSyncWithRawParamsInVM({
+          formulaSpecification,
+          params,
+          bundleSourceMapPath,
+          bundlePath,
+          executionContext,
+        })
+      : await executeFormulaOrSyncWithRawParams({formulaSpecification, params, manifest, executionContext});
+
+    result.push(...response.result);
+    if (contextOptions?.syncOptions?.includePermissions) {
+      // Denormalize the sync result before passing back into executeGetPermissions
+      const schema =
+        executionContext.sync.schema ??
+        findSyncFormula(manifest, formulaSpecification.formulaName, executionContext.authenticationName)?.schema;
+      const denormalizedSyncResult = untransformBody(response.result, schema);
+
+      rowAccessDefinitions.push(
+        ...(await getPermissionsForRows({
+          rows: denormalizedSyncResult,
+          formulaSpecification,
+          params,
+          manifest,
+          executionContext,
+          vm,
+          bundleSourceMapPath,
+          bundlePath,
+        })),
+      );
+    }
+    executionContext.sync.continuation = response.continuation;
+    iterations++;
+  } while (executionContext.sync.continuation && result.length < maxRows);
+  if (result.length > maxRows) {
+    result = result.slice(0, maxRows);
+  }
+  if (rowAccessDefinitions.length > maxRows) {
+    rowAccessDefinitions = rowAccessDefinitions.slice(0, maxRows);
```

Comment:
Hmm, I don't think I'd do this. The relationship of permissions to rows is many:1 so we could easily exceed the row count with permissions and I don't think we should arbitrarily truncate.

I think maxRows is just meant to be a safety stop for the sync loop, so I think we can ignore it here.